### PR TITLE
[List][RLMArray] Allow moving objects from one index to another

### LINKED ISSUE #2242: [RLMArray][List] Implement Move method
To support drag'n'drop reordering in lists we need a move method.

This is needed by sync as a move is semantically very different from a delete and insert. It is also important to avoid triggering cascading deletes when we start supporting that.
```obj-c
- (void)moveObjectFromIndex:(NSUInteger)from toIndex:(NSUInteger)to;
```

### PR DESCRIPTION
**Pull request by segiddins**
Closes #2242.

Depends upon https://github.com/realm/realm-core/pull/955.

### CODE REVIEW

**Comment by segiddins** [comment]
Failure is unrelated.

\c @jpsim

File: Realm/RLMArray.h [hunk 1]
```diff
@@ -178,13 +178,26 @@ RLM_ASSUME_NONNULL_BEGIN
...
  @warning This method can only be called during a write transaction.
- 
+
  @param index       The array index of the object to be replaced.
  @param anObject    An object (of the same type as returned from the objectClassName selector).
  */
 - (void)replaceObjectAtIndex:(NSUInteger)index withObject:(RLMObjectArgument)anObject;
 
 /**
+ Moves the object at from the given source index to the given destination index.
```
Comment (jpsim) [inline Realm/RLMArray.h]: > Moves the object from the given source index to the given destination index.

Same change needs to be applied on all 3 versions of these docs.

File: Realm/RLMArray.h [hunk 2]
```diff
@@ -178,13 +178,26 @@ RLM_ASSUME_NONNULL_BEGIN
...
+
+ Throws an exception when called with an index greater than or equal to the
```
Comment (jpsim) [inline Realm/RLMArray.h]: > Throws an exception when called with an index exceeding the bounds of this array.

**Comment by jpsim** [comment]
:+1:

**Pull request merged**